Streetcode/Streetcode.BLL/MediatR/Media/Image/GetByStreetcodeId/GetImageByStreetcodeIdHandler.cs

```diff
@@ -18,37 +20,47 @@ public class GetImageByStreetcodeIdHandler : IRequestHandler<GetImageByStreetcod
     private readonly IRepositoryWrapper _repositoryWrapper;
     private readonly ILoggerService _logger;
     private readonly IStringLocalizer<CannotFindSharedResource> _stringLocalizerCannotFind;
+    private readonly ICacheService _cacheService;
 
-    public GetImageByStreetcodeIdHandler(IRepositoryWrapper repositoryWrapper, IMapper mapper, IBlobService blobService, ILoggerService logger, IStringLocalizer<CannotFindSharedResource> stringLocalizerCannotFind)
+    public GetImageByStreetcodeIdHandler(IRepositoryWrapper repositoryWrapper, IMapper mapper, IBlobService blobService, ILoggerService logger, IStringLocalizer<CannotFindSharedResource> stringLocalizerCannotFind, ICacheService cacheService)
     {
         _repositoryWrapper = repositoryWrapper;
         _mapper = mapper;
         _blobService = blobService;
         _logger = logger;
         _stringLocalizerCannotFind = stringLocalizerCannotFind;
+        _cacheService = cacheService;
     }
 
     public async Task<Result<IEnumerable<ImageDTO>>> Handle(GetImageByStreetcodeIdQuery request, CancellationToken cancellationToken)
     {
-        var images = (await _repositoryWrapper.ImageRepository
-            .GetAllAsync(
-            f => f.Streetcodes.Any(s => s.Id == request.StreetcodeId),
-            include: q => q.Include(img => img.ImageDetails))).OrderBy(img => img.ImageDetails?.Alt);
-
-        if (images is null || images.Count() == 0)
-        {
-            string errorMsg = _stringLocalizerCannotFind["CannotFindAnImageWithTheCorrespondingStreetcodeId", request.StreetcodeId].Value;
-            _logger.LogError(request, errorMsg);
-            return Result.Fail(new Error(errorMsg));
-        }
-
-        var imageDtos = _mapper.Map<IEnumerable<ImageDTO>>(images);
-
-        foreach (var image in imageDtos)
-        {
-            image.Base64 = _blobService.FindFileInStorageAsBase64(image.BlobName);
-        }
-
-        return Result.Ok(imageDtos);
+        string cacheKey = $"ImageCache_{request.StreetcodeId}";
```

Comment:
Let's move this cache keys to some standardized solution, we shouldn't hardcode this key here and in every method we need this